### PR TITLE
feat: Updated adxl345 sensor driver to configuration via devicetree file. 

### PR DESCRIPTION
**Pull request by whati001**
Followup PR of: #66046 

Updated/rewritten adxl345 sensor code to support devicetree parameters. I have tested the code with an nrf52840-dk and a adxl345 connected via I2C.

### CODE REVIEW

File: drivers/sensor/adxl345/adxl345.c [hunk 1]
```diff
@@ -63,50 +63,59 @@ static int adxl345_reg_access_spi(const struct device *dev, uint8_t cmd, uint8_t
...
-static inline int adxl345_reg_read(const struct device *dev, uint8_t addr, uint8_t *data,
-				   uint8_t len)
+static int adxl345_reg_read(const struct device *dev, uint8_t addr, uint8_t *data, uint8_t len)
 {
-
 	return adxl345_reg_access(dev, ADXL345_READ_CMD, addr, data, len);
 }
 
-static inline int adxl345_reg_write_byte(const struct device *dev, uint8_t addr, uint8_t val)
+int adxl345_reg_write_byte(const struct device *dev, uint8_t addr, uint8_t val)
```
Comment (yperess) [inline drivers/sensor/adxl345/adxl345.c:84]: nit: leave `static`

File: drivers/sensor/adxl345/adxl345.c [hunk 2]
```diff
@@ -63,50 +63,59 @@ static int adxl345_reg_access_spi(const struct device *dev, uint8_t cmd, uint8_t
...
 {
 	return adxl345_reg_write(dev, addr, &val, 1);
 }
 
-static inline int adxl345_reg_read_byte(const struct device *dev, uint8_t addr, uint8_t *buf)
-
+int adxl345_reg_read_byte(const struct device *dev, uint8_t addr, uint8_t *buf)
```
Comment (yperess) [inline drivers/sensor/adxl345/adxl345.c:89]: nit: leave `static`

File: drivers/sensor/adxl345/adxl345.c [hunk 3]
```diff
@@ -116,40 +125,36 @@ static int adxl345_read_sample(const struct device *dev,
...
+		samples_count = 1;
+	} else {
+		rc = adxl345_reg_read_byte(dev, ADXL345_FIFO_STATUS_REG, &samples_count);
+		if (rc < 0) {
+			LOG_ERR("Failed to read FIFO status rc = %d\n", rc);
+			return rc;
+		}
 	}
 
 	__ASSERT_NO_MSG(samples_count <= ARRAY_SIZE(data->bufx));
```
Comment (yperess) [inline drivers/sensor/adxl345/adxl345.c:160]: I noticed the following in the datasheet:

> there must be at least 5 µs between the end of reading
> the data registers and the start of a new read of the FIFO or a read
> of the FIFO_STATUS register (Address 0x39). The end of reading
> a data register is signified by the transition from Register 0x37 to
> Register 0x38 or by the CS pin going high.

The loop below might need to be tweaked to verify this delay.

**Review by yperess — CHANGES_REQUESTED** [review]

**Review by teburd — DISMISSED** [review]

File: drivers/sensor/adxl345/adxl345.c (hunk 1 above)
Comment (whati001) [reply drivers/sensor/adxl345/adxl345.c:84]: added `static`

File: drivers/sensor/adxl345/adxl345.c (hunk 3 above)
Comment (whati001) [reply drivers/sensor/adxl345/adxl345.c:160]: The loop will be removed via #67499 . Added temporarily a sleep for 5 us.

File: drivers/sensor/adxl345/adxl345.c [hunk 4]
```diff
@@ -255,34 +353,41 @@ static int adxl345_init(const struct device *dev)
...
+                                                                                                   \
+	static const struct adxl345_dev_config adxl345_config_##inst =                             \
+		COND_CODE_1(DT_INST_ON_BUS(inst, spi), (ADXL345_CONFIG_SPI(inst)),                 \
+			    (ADXL345_CONFIG_I2C(inst)));                                           \
+                                                                                                   \
+	SENSOR_DEVICE_DT_INST_DEFINE(inst, adxl345_init, NULL, &adxl345_data_##inst,               \
+				     &adxl345_config_##inst, POST_KERNEL,                          \
+				     CONFIG_SENSOR_INIT_PRIORITY, &adxl345_api_funcs);
+
+DT_INST_FOREACH_STATUS_OKAY(ADXL345_DEFINE)
```
Comment (MaureenHelm) [inline drivers/sensor/adxl345/adxl345.c]: This looks like a lot of whitespace noise. Please keep major whitespace reformatting changes in separate commits from functional changes

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thanks for your contribution!

Please address compliance failures and review feedback by amending commits instead of adding fixup commits.

File: drivers/sensor/adxl345/adxl345.c (hunk 4 above)
Comment (whati001) [reply drivers/sensor/adxl345/adxl345.c]: Refactored code to reduce whitespace noise.

**Comment by whati001** [comment]
Updated code and normally fixed compliance checks. Please apologize, but I am not sure why the last three checks are not running.

**Comment by whati001** [comment]
@MaureenHelm please provide an update

File: drivers/sensor/adxl345/adxl345.c [hunk 5]
```diff
@@ -63,50 +63,59 @@ static int adxl345_reg_access_spi(const struct device *dev, uint8_t cmd, uint8_t
...
 
-static inline int adxl345_reg_write_byte(const struct device *dev, uint8_t addr, uint8_t val)
+static int adxl345_reg_write_byte(const struct device *dev, uint8_t addr, uint8_t val)
 {
 	return adxl345_reg_write(dev, addr, &val, 1);
 }
 
-static inline int adxl345_reg_read_byte(const struct device *dev, uint8_t addr, uint8_t *buf)
-
+static int adxl345_reg_read_byte(const struct device *dev, uint8_t addr, uint8_t *buf)
```
Comment (MaureenHelm) [inline drivers/sensor/adxl345/adxl345.c:89]: These changes aren't related to `feat: Add devicetree support for adxl345 sensor`. Please revert

File: drivers/sensor/adxl345/adxl345.c [hunk 6]
```diff
@@ -116,40 +125,37 @@ static int adxl345_read_sample(const struct device *dev,
...
-		return rc;
+
+	if (data->fifo_mode == ADXL345_FIFO_BYPASS) {
+		samples_count = 1;
+	} else {
+		rc = adxl345_reg_read_byte(dev, ADXL345_FIFO_STATUS_REG, &samples_count);
+		if (rc < 0) {
+			LOG_ERR("Failed to read FIFO status rc = %d\n", rc);
+			return rc;
+		}
```
Comment (MaureenHelm) [inline drivers/sensor/adxl345/adxl345.c:158]: Also not related to devicetree. Please revert

File: drivers/sensor/adxl345/adxl345.c [hunk 7]
```diff
@@ -163,13 +169,14 @@ static int adxl345_sample_fetch(const struct device *dev,
 		data->bufx[s] = sample.x;
 		data->bufy[s] = sample.y;
 		data->bufz[s] = sample.z;
+
+		k_usleep(ADXL345_SAMPLE_SLEEP_TIME_US);
```
Comment (MaureenHelm) [inline drivers/sensor/adxl345/adxl345.c:173]: Same

File: dts/bindings/sensor/adi,adxl345-common.yaml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+description: ADXL345 3-axis I2C accelerometer
+
+include: [sensor-device.yaml]
+
+properties:
+  frequency:
+    type: int
+    default: 100
+    description: |
+          Accelerometer sampling frequency (ODR). Default is 100Hz.
```
Comment (MaureenHelm) [inline dts/bindings/sensor/adi,adxl345-common.yaml:13]: Need to justify the default value in the description. Is it a register reset value?

File: dts/bindings/sensor/adi,adxl345-common.yaml [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+      - 50
+      - 100
+      - 200
+      - 400
+
+  range:
+    type: int
+    default: 16
+    description: |
+      Accelerometer sampling range in G
```
Comment (MaureenHelm) [inline dts/bindings/sensor/adi,adxl345-common.yaml:27]: Need to justify the default value in the description. Is it a register reset value?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: dts/bindings/sensor/adi,adxl345-common.yaml (hunk 9 above)
Comment (whati001) [reply dts/bindings/sensor/adi,adxl345-common.yaml]: Updated default value to reset value and updated description text.

File: dts/bindings/sensor/adi,adxl345-common.yaml (hunk 8 above)
Comment (whati001) [reply dts/bindings/sensor/adi,adxl345-common.yaml]: Added justification.

File: drivers/sensor/adxl345/adxl345.c (hunk 5 above)
Comment (whati001) [reply drivers/sensor/adxl345/adxl345.c]: reverted white space noise as good as possible.

File: drivers/sensor/adxl345/adxl345.c (hunk 6 above)
Comment (whati001) [reply drivers/sensor/adxl345/adxl345.c:164]: This change is needed to fetch samples properly if FIFO is disabled.

Because if FIFO is disabled, reading from ADXL345_FIFO_STATUS_REG returns 0.
If sample_count is zero, the driver will never fetch a sample within the loop body.
image

Comment (whati001) [reply drivers/sensor/adxl345/adxl345.c:164]: This behavior will be fixed with #67499

**Comment by whati001** [comment]
@MaureenHelm please apologize to bother you again, but I have now reverted all whitespace issues as good as possible and one change is mandatory to ensure functionality if FIFO is disabled (possible now via DT setup).
Maybe you can catch some time to review one more time :)

**Comment by whati001** [comment]
Rebased on current main

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
> @MaureenHelm please apologize to bother you again, but I have now reverted all whitespace issues as good as possible and one change is mandatory to ensure functionality if FIFO is disabled (possible now via DT setup).
Maybe you can catch some time to review one more time :)

Thank you, but these changes need to be amended to the original commits. In other words, don't introduce all the whitespace changes in one commit and then revert them in another commit; avoid introducing the whitespace changes in the first place. Similarly, amend the commit that introduces the new devicetree properties to include their default justifications.

**Comment by whati001** [comment]
Hi @MaureenHelm 

thank you very much for your feedback and please apologize the circumstances.
As requested, I have squashed all the commits and force pushed to includes everything within a single commit without "reverting" commits.

**Comment by whati001** [comment]
Fixed compliance issues, please review again @MaureenHelm
